src/clients/sync.rs

```diff
@@ -24,7 +24,8 @@ pub(crate) async fn sync_db(
     store: &LocalStore,
     cache: &Cache,
     status: fn(&str),
-) -> Result<(), Error> {
+    existing_addresses: Option<&[Address]>,
```

Comment:
What if you remove the Option and just use an empty vec for None? It should simplify a lot